src/librustc_plugin/registry.rs

```diff
@@ -26,6 +25,8 @@ pub struct Registry<'a> {
     /// from the plugin registrar.
     pub sess: &'a Session,
 
+    pub lint_store: &'a mut LintStore,
```

Comment:
Nit: doc-comment, I think something simple like this would suffice

---

The `LintStore` permits plugins to register new lints.